lib/services/nodes-api-service.js

```diff
@@ -509,6 +509,26 @@ function nodeApiServiceFactory(
     };
 
     /**
+     * Remove the tag from a list of nodes
+     * @param  {String}     tagName
+     * @return {Promise}
+     */
+    NodeApiService.prototype.masterDelTagById = function(tagName) {
```

Comment:
The tagName is enforced as unique in the code.  The model should be updated to add the unique tag.

The term id is used to refer interchangeably to nodeid, graphid, taskid, etc throughout the code regardless of them being mongo id, uuid, etc.  In this case, 'id' is being used to refer correctly to the tag's primary key (which happens to be a stringid)